onecodex/viz/_bargraph.py

```diff
@@ -0,0 +1,124 @@
+            Text label at the top of the plot.
+        xlabel : `string`, optional
+            Text label along the horizontal axis.
+        ylabel : `string`, optional
+            Text label along the vertical axis.
+        tooltip : `string` or `list`, optional
+            A string or list containing strings representing metadata fields. When a point in the
+            plot is hovered over, the value of the metadata associated with that sample will be
+            displayed in a modal.
+    """
+
+    def plot_bargraph(
+        self,
+        rank="auto",
+        normalize="auto",
+        threshold="auto",
+        top_n="auto",
+        title=None,
+        xlabel=None,
+        ylabel=None,
+        tooltip=None,
+        return_chart=False,
+    ):
+
+        if rank is None:
+            raise OneCodexException(
+                "Please specify a rank or 'auto' to choose automatically"
+            )
+
+        if not (threshold or top_n):
+            raise OneCodexException("Please specify at least one of: threshold, top_n")
+
+        if top_n == "auto" and threshold == "auto":
+            top_n = 10
+            threshold = None
+        elif top_n == "auto" and threshold != "auto":
+            top_n = None
+        elif top_n != "auto" and threshold == "auto":
+            threshold = None
+
+        df = self.to_df(
+            rank=rank,
+            normalize=normalize,
+            top_n=top_n,
+            threshold=threshold,
+            table_format="long",
+        )
+
+        if tooltip:
+            if not isinstance(tooltip, list):
+                tooltip = [tooltip]
+        else:
+            tooltip = []
+
+        # takes metadata columns and returns a dataframe with just those columns
+        # renames columns in the case where columns are taxids
+        magic_metadata, magic_fields = self._metadata_fetch(tooltip)
+
+        for f in tooltip:
+            df[magic_fields[f]] = magic_metadata[magic_fields[f]][
+                df["classification_id"]
+            ].tolist()
+
+        # add sample filenames
+        df["display_name"] = self.metadata["_display_name"][
+            df["classification_id"]
+        ].tolist()
+
+        # add taxa names
+
+        df["tax_name"] = [
+            "{} ({})".format(self.taxonomy["name"][t], t)
+            if t in self.taxonomy["name"]
+            else t
+            for t in df["tax_id"]
+        ]
+
+        #
+        # TODO: how to sort bars in bargraph
+        # - abundance (mean across all samples)
+        # - parent taxon (this will require that we make a few assumptions
+        # about taxonomic ranks but as all taxonomic data will be coming from
+        # OCX this should be okay)
+        #
+
+        chart = (
+            alt.Chart(df)
+            .mark_bar()
+            .encode(
+                x=alt.X("display_name", axis=alt.Axis(title=xlabel)),
+                y=alt.Y("readcount_w_children", axis=alt.Axis(title=ylabel)),
```

Comment:
Let's check if `xlabel` or `ylabel` are `None` and provide a default behavior? I don't remember... if we set `title=None` does that mean there's no axis labels?

Reply:
Labels will be the same as if `axis=` wasn't specified. Altair is inconsistent with this. For example, setting `chart.properties(title=None)` will result in an error.

<img width="505" alt="screen shot 2019-02-15 at 11 23 26 am" src="https://user-images.githubusercontent.com/115904/52879505-24526f80-3114-11e9-9df5-aedf0b53f351.png">